app/controllers/sessions_controller.rb

```diff
@@ -1,12 +1,18 @@
 class SessionsController < ApplicationController
+  before_action :find_by_email, only: :create
+
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user&.authenticate(params[:session][:password])
-      log_in user
-      remember user
-      redirect_back_or user
+    if @user&.authenticate(params[:session][:password])
+      if @user.activated?
+        log_in @user
+        remember @user
+        redirect_back_or @user
+      else
+        flash[:warning] = t ".activate_fail"
+        redirect_to root_url
```

Comment:
```suggestion
        redirect_to root_path
```
trong các controller thì dùng path